content/docs/guides/multi-target-exporter.md

```diff
@@ -0,0 +1,521 @@
+---
+title: Understanding and using the multi-target exporter pattern
+---
+
+# Understanding and using the multi-target exporter pattern
+
+This guide will introduce you to the multi-target exporter pattern. To achieve this we will:
+
+* describe the multi-target exporter pattern and why it is used,
+* run the [blackbox](https://github.com/prometheus/blackbox_exporter) exporter as an example of the pattern,
+* configure a custom query module for the blackbox exporter,
+* let the blackbox exporter run basic metric queries against the Prometheus [website](https://prometheus.io),
+* examine a popular pattern of configuring Prometheus to scrape exporters using relabeling.
+
+## The multi-target exporter pattern?
+
+By multi-target [exporter](/docs/instrumenting/exporters/) pattern we refer to a specific design, in which:
+
+* the exporter will get the target’s metrics via a network protocol.
+* the exporter does not have to run on the machine the metrics are taken from.
+* the exporter gets the targets and a query config string as parameters of Prometheus’ GET request.
+* the exporter subsequently starts the scrape after getting Prometheus’ GET requests and once it is done with scraping.
+* the exporter can query multiple targets.
+
+This pattern is only used for certain exporters, namely the [blackbox](https://github.com/prometheus/blackbox_exporter) and the [SNMP exporter](https://github.com/prometheus/snmp_exporter).
+
+The reason is that we either can’t run the exporter on the targets, e.g. network gear speaking SNMP, or that we are explicitly interested in the distance, e.g. latency and reachability of a website from a specific point outside of our network.
+
+## Running multi-target exporters
+
+Multi-target exporters are flexible regarding their environment and can be run in many ways. As regular programs, in containers, as background services, on baremetal, virtual machines. Because they are queried and do query over network they do need appropriate open ports. Otherwise they are frugal.
+
+Now let’s try it out for yourself!
+
+Use [Docker](https://www.docker.com/) to start a blackbox exporter container running by running this in a terminal:
+
+```bash
+sudo docker run -p 9115:9115 prom/blackbox-exporter
```

Comment:
Why the sudo?

Reply:
Because this docker command does not work without using sudo.

Reply:
I don't understand why this should require sudo, this sounds like something odd with your setup rather than a general thing about docker.

Reply:
I tested this on a very vanilla Ubuntu Linux like many potential readers of the guide would use and it needs sudo. What more can I tell you about it?

Reply:
The command works fine for me without sudo, and I'm also on vanilla Ubuntu. I'm guessing your user isn't a member of the docker group?

In generally let's not recommend use of sudo when it's not 100% required.

Reply:
I haven't tried it yet but docker on Mac / Windows is strongly tied to the _user_ so I expect `sudo docker` tp either blow up spectacularly or misbehave subtly. let's give the command without it, and mention in a separate sentence that in some setups you may need sudo?

Reply:
Okay, that sounds like a reasonable approach between best practices and raising the bar to entry. I’ll add it.